### ISSUE TITLE
No Error Check in Debugger_Threat_PyDbg in Spawn_target

### ISSUE THREAD

**Issue by KevinCooper** [jtpereyda/boofuzz#237]
See the below for the crash.
```
C:\Users\user\Desktop\boofuzz>python process_monitor.py --proc-name program.exe
[09:29.14] Process Monitor PED-RPC server initialized:
[09:29.14]       listening on:  0.0.0.0:26002
[09:29.14]       crash file:    C:\Users\user\Desktop\boofuzz\boofuzz-crash-bin
[09:29.14]       # records:     0
[09:29.14]       proc name:     program.exe
[09:29.14]       log level:     1
[09:29.14] awaiting requests...
[09:29.25] Starting target...
[09:29.25] starting target process
[09:29.25] done. target up and running, giving it 5 seconds to settle in.
Exception in thread 1550330965:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\Users\user\Desktop\boofuzz\boofuzz\utils\debugger_thread_pydbg.py", line 118, in run
    self.spawn_target()
  File "C:\Users\user\Desktop\boofuzz\boofuzz\utils\debugger_thread_pydbg.py", line 112, in spawn_target
    self.pid = self._process.pid
AttributeError: 'NoneType' object has no attribute 'pid'
```

I believe that this is caused by non-implemented TODO on line 101 of `debugger_thread_pydbg.py`.  Since there are no commands in `self.start_commands`, then self._process never gets set.  This causes the `NoneType` error on line 112.

`# TODO checks: debugger thread already active; process already started; no start_commands`

**Comment by jtpereyda** [comment]
@KevinCooper Thank you for the bug report! It seems like the correct behavior in this case would be to search for a process based on the `--proc-name` command line argument. Does that sound right?

**Comment by jtpereyda** [comment]
@KevinCooper I attached a pull request to fix but am not able to test atm. If you have your test rig up still, feel free to try it out. If someone can confirm it works I can merge.

**Comment by KevinCooper** [comment]
PR fixes the issue.  Thank you!

**Comment by KevinCooper** [comment]
Do note, since the **while True** gets changed to the **for loop**, it is possible for the following error to occur if the process_name does not exist.
```
[10:26.43] debugger thread-1550636803 looking for process name: program.exe
Exception in thread 1550636803:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\Users\user\Desktop\boofuzz\boofuzz\utils\debugger_thread_pydbg.py", line 122, in run
    self.watch()
  File "C:\Users\user\Desktop\boofuzz\boofuzz\utils\debugger_thread_pydbg.py", line 158, in watch
    self.process_monitor.log("debugger thread-%s found match on pid %d" % (self.getName(), self.pid))
TypeError: %d format: a number is required, not NoneType
```

**Comment by jtpereyda** [comment]
@KevinCooper Thanks for the catch; should be fixed now after #244.
